src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/Wizard.vue

```diff
@@ -16,7 +16,39 @@
     <v-stepper-window v-model="wizardStore.step">
       <v-stepper-window-item v-for="(step, index) in Object.values(wizard.steps)" :key="step.stage" :value="index + 1">
         <v-container>
-          <h3>{{ step.title }}</h3>
+          <v-row class="justify-space-between">
+            <v-col cols="auto">
+              <h3>{{ step.title }}</h3>
+            </v-col>
+            <v-col v-if="wizardStore.currentStepStage === 'Review'" cols="auto">
+              <ConfirmationDialog
+                v-if="!isWizardDataValid"
+                :config="{ cancelButtonText: 'Cancel', acceptButtonText: 'Yes', title: 'Print Confirmation', customButtonVariant: 'text' }"
+                @accept="printPage"
+              >
+                <template #activator>
+                  <v-icon color="secondary" icon="mdi-printer-outline" class="mr-2"></v-icon>
+                  <a class="small">Print Preview</a>
+                </template>
+                <template #confirmation-text>
+                  <p>
+                    Your Application contains missing data and/or invalid data.
+                    <br />
+                    The printed preview will show which sections are incomplete
+                  </p>
+                  <br />
+                  <p><b>Are you sure you want to proceed?</b></p>
+                </template>
+              </ConfirmationDialog>
+              <v-btn v-if="isWizardDataValid" variant="text" onclick="window.print()">
+                <v-row align="center" justify="end">
+                  <v-icon color="secondary" icon="mdi-printer-outline" class="mr-2"></v-icon>
+                  <a class="small">Print Preview</a>
+                </v-row>
+              </v-btn>
```

Comment:
just food for thought. We could abstract out the v-if and send in a boolean to ConfirmationDialog to see if confirmation is needed or not.

Reply:
Implemented the changes.